packages/core/src/helpers/extractors.ts

```diff
@@ -105,31 +95,29 @@ function timestampExtractor(nDelimitedData: Field[]) {
  * Extracts date of birth and gender information from delimited data,
  * calculates the person’s current age based on the current date.
  *
- * Rows: 10051
+ * Rows: 9416
  *
  * @param {Field[]} nDelimitedData - The delimited input data array.
- * @param {Field[]} delimiterIndices - Array of indices marking field positions.
  * @param {Field} currentYear - Current year as a Field.
  * @param {Field} currentMonth - Current month as a Field.
  * @param {Field} currentDay - Current day as a Field.
  * @returns {[Field, Field]} A tuple: [calculated age, extracted gender].
  */
 function ageAndGenderExtractor(
   nDelimitedData: Field[],
-  delimiterIndices: Field[],
   currentYear: Field,
   currentMonth: Field,
   currentDay: Field
 ) {
-  let ageData: Field[] = []
-  const startIndex = delimiterIndices[DOB_POSITION - 1]
+  const ageData: Field[] = []
+  const startIndex = Provable.witness(Field, () => {
+    return  nDelimitedData.findIndex((value) => value.toBigInt() === BigInt(DELIMITER_POSITION.DOB * 255))
+  })
```

Comment:
Before using startIndex w/ `selectSubarray` and `searchElement`, I would recommend constraining it like below:
```
startIndex.assertGreaterThanOrEqual(Field(0))
startIndex.assertLessThan(Field(DATA_ARRAY_SIZE))
```